extend-string.js

```diff
@@ -0,0 +1,126 @@
+/* Checks for vowel(s) in a string */
+String.prototype.hasVowels = function() {
+  let regEx = /[aeiou]/g; // pattern to search for vowels 
+  return regEx.test(this)? true : false
+}
+
+/* Converts each string character to uppercase */
+String.prototype.toUpper = function() {
+  let result = ''; // empty string variable to store result
+  for(index = 0; index < this.length; index++){ 
+    let characterCode = this.charCodeAt(index); // get the character code of each character
+    if(characterCode >= 97 && characterCode <= 122){ //check if character code is in lowercase using ascii code
+      characterCode -= 32 // subtract 32 from the character code to get its uppercase equivalent code
+    } 
+    result += String.fromCharCode(characterCode); //appending each of the transformed character to the result variable
+  }
+  return result // return the 
+}
+
+/* Converts each string character to lowercase */
+String.prototype.toLower = function() {
+  let result = ''; // empty string variable to store result
+  for(index = 0; index < this.length; index++){ 
+    let characterCode = this.charCodeAt(index); // get the character code of each character
+    if(characterCode >= 65 && characterCode <= 90){ //check if character code is in uppercase using ascii code
+      characterCode += 32 // add 32 to the character code to get its lowercase equivalent code
+    } 
+    result += String.fromCharCode(characterCode); //appending each of the transformed character to the result variable
+  }
+  return result
+};
+
+/* Converts first letter of a string to uppercase */
+String.prototype.ucFirst = function() {
+  if(/[A-Z]/.test(this)) return 'First character already capitalized'
+  return this.charAt(0).toUpper() + this.slice(1) //converts first letter to upper and joins it to the rest of the string characters
+}
+
+/* Checks if a string is a question */
+String.prototype.isQuestion = function() {
+  let regEx = /\w+\?$/g; // pattern to search for strings ending with ?
+  return regEx.test(this)? true: false
+}
+
+/* Pulls out all the words in a string */
+String.prototype.words = function() {
+  let regEx = /[\w-]+/g; // pattern to match a word(s)
+  return this.match(regEx);
+}
+
+/* Counts all the words in a string */
+String.prototype.wordCount = function() {
+  return this.words().length // gets the number of words in string
+}
+
+
+/*Gets currency representation of a string*/
+String.prototype.toCurrency = function() {
+  let regEx = /\d(?=\d{3}\.)/g; //pattern to match digit immediately followed by 3 digits
```

Comment:
try the consideration that the stringed number given may not be a decimal. i.e. 111111111 or decimals longer than 6 digits 1111111.11